app/services/reinstate_declined_offer.rb

```diff
@@ -0,0 +1,33 @@
+class ReinstateDeclinedOffer
+  def initialize(course_choice:, zendesk_ticket:)
+    @course_choice = course_choice
+    @zendesk_ticket = zendesk_ticket
+  end
+
+  def save!
+    reset_all_dbd
+
+    @course_choice.update!(
+      status: 'offer',
+      declined_at: nil,
+      declined_by_default: false,
+      decline_by_default_at: 10.business_days.from_now,
+      audit_comment: "Reinstate offer Zendesk request: #{@zendesk_ticket}",
+    )
+  end
+
+private
+
+  def reset_all_dbd
+    choices_to_reset.each do |choice|
+      choice.update!(
+        decline_by_default_at: 10.business_days.from_now,
```

Comment:
Could we use the `TimeLimitCalculator` to set DBD? It is 10 business days, but it's not guaranteed to be